src/setting.rs

```diff
@@ -201,636 +182,271 @@ pub enum SettingData {
-        let (input, lcd_brightness) = LCDBrightness::parse(input)?;
-        let (input, quantize) = Quantize::parse(input)?;
-        let (input, auto_cue_level) = AutoCueLevel::parse(input)?;
-        let (input, language) = Language::parse(input)?;
-        let (input, unknown2) = nom::number::complete::u8(input)?;
-        let (input, jog_ring_brightness) = JogRingBrightness::parse(input)?;
-        let (input, jog_ring_indicator) = JogRingIndicator::parse(input)?;
-        let (input, slip_flashing) = SlipFlashing::parse(input)?;
-        let (input, unknown3) = nom::bytes::complete::take(3usize)(input)?;
-        let unknown3 = unknown3.to_vec();
-        let (input, disc_slot_illumination) = DiscSlotIllumination::parse(input)?;
-        let (input, eject_lock) = EjectLock::parse(input)?;
-        let (input, sync) = Sync::parse(input)?;
-        let (input, play_mode) = PlayMode::parse(input)?;
-        let (input, quantize_beat_value) = QuantizeBeatValue::parse(input)?;
-        let (input, hotcue_autoload) = HotCueAutoLoad::parse(input)?;
-        let (input, hotcue_color) = HotCueColor::parse(input)?;
-        let (input, unknown4) = nom::number::complete::be_u16(input)?;
-        let (input, needle_lock) = NeedleLock::parse(input)?;
-        let (input, unknown5) = nom::number::complete::be_u16(input)?;
-        let (input, time_mode) = TimeMode::parse(input)?;
-        let (input, jog_mode) = JogMode::parse(input)?;
-        let (input, auto_cue) = AutoCue::parse(input)?;
-        let (input, master_tempo) = MasterTempo::parse(input)?;
-        let (input, tempo_range) = TempoRange::parse(input)?;
-        let (input, phase_meter) = PhaseMeter::parse(input)?;
-        let (input, unknown6) = nom::number::complete::be_u16(input)?;
-        let data = Self::MySetting {
-            unknown1,
-            on_air_display,
-            lcd_brightness,
-            quantize,
-            auto_cue_level,
-            language,
-            unknown2,
-            jog_ring_brightness,
-            jog_ring_indicator,
-            slip_flashing,
-            unknown3,
-            disc_slot_illumination,
-            eject_lock,
-            sync,
-            play_mode,
-            quantize_beat_value,
-            hotcue_autoload,
-            hotcue_color,
-            unknown4,
-            needle_lock,
-            unknown5,
-            time_mode,
-            jog_mode,
-            auto_cue,
-            master_tempo,
-            tempo_range,
-            phase_meter,
-            unknown6,
-        };
-        Ok((input, data))
-    }
 }
 
 /// Found at "PLAYER > DJ SETTING > PLAY MODE / AUTO PLAY MODE" of the "My Settings" page in the
 /// Rekordbox preferences.
-#[derive(Debug)]
+#[derive(Debug, PartialEq)]
+#[binrw]
+#[brw(repr = u8)]
 pub enum PlayMode {
     /// Named "CONTINUE / ON" in the Rekordbox preferences.
-    Continue,
+    Continue = 0x80,
     /// Named "SINGLE / OFF" in the Rekordbox preferences.
```

Comment:
```rust
#[brw(repr = i8)]
pub enum PlayMode {
    /// Named "CONTINUE / ON" in the Rekordbox preferences.
    Continue = i8::MIN,
```
IMO also a viable option of expressing the same thing. I'm not sure why the values start at `0x80`. Could either be that the enums start at the lowest value and are signed (for whatever reason), or they just carry some extra info in their highest bit. Idk, feel free to reject this suggestion, I just wanted to throw the thought out there.

Reply:
It would be interesting if we had some files from older rekordbox versions. I have a hunch that the highest bit indicates if a field is in use.

Reply:
Pioneer still hosts older rekordbox versions: https://rekordbox.com/en/support/faq/about-rekordbox-5/ (see the "I want to use previous rekordbox ver. 5." Question).

Reply:
can you elaborate what you mean with "if a field is in use"? In use by what?

Reply:
> can you elaborate what you mean with "if a field is in use"? In use by what?

A bunch of fields/bytes are just 0x00. But every actual value is > 0x80. I guess that the 0x00 bytes are reserved for future use. 

If a new CDJ comes out, Pioneer might decide to add new config options. If you make a fresh export using the latest Rekordbox version, the generated DAT will contain the selected value for that new config option. 

Now let's say someone is still using RB5, then thr generated config files will not contain that config option and the field will just be 0x00 ("not set"). The CDJ can detect that and use a sensible default instead.

Reply:
Sounds plausible, but on the other hand, I'd be very surprised if Pioneer cared about any sort of backwards compatibility with RB5. Its more plausible that they want to maintain compatibility with old CDJs, but your theory does not work out for preserving compatibility in that direction.

Reply:
For the other direction this is not necessary. When the CDJ is released, it supports a specific set of options. If Pioneer adds more options, the CDJ can just ignore these bytes.

Reply:
> I'd be very surprised if Pioneer cared about any sort of backwards compatibility with RB5.

It would kind of suck if you plug your prepared USB drive into the CDJs at a venue and it doesn't work because the drive was created using an old RB version.

>  Its more plausible that they want to maintain compatibility with old CDJs, but your theory does not work out for preserving compatibility in that direction.

For the other direction this is not necessary. The firmware of the CDJ knows the set of options it supports, and can safely ignore all other bytes in the file, even if more recent rekordbox versions start using them for new options.

Reply:
> It would kind of suck if you plug your prepared USB drive into the CDJs at a venue and it doesn't work because the drive was created using an old RB version.

I fully agree, but Pioneer is a profit-driven company that sells not only their hardware products, but also their rekordbox software, so I'd be pretty surprised if they actually have the decency to support RB5 when they could just force their consumers to subscribe to RB6. After all, they essentially have a monopoly in the professional DJ space.

Reply:
> For the other direction this is not necessary. The firmware of the CDJ knows the set of options it supports, and can safely ignore all other bytes in the file, even if more recent rekordbox versions start using them for new options.

That is only possible if the binary file is structured appropriately. I have not taken my time to understand it to the point where I can evaluate that, but I guess you can.